app/javascript/packs/flagUserModal.js

```diff
@@ -0,0 +1,137 @@
+import { request } from '../utilities/http';
+
+export default function initializeFlagUserModal(articleAuthorId) {
+  const flagUserModalHTML = `
+<div class="crayons-modal crayons-modal--s absolute flag-user-modal">
+  <div class="crayons-modal__box">
+    <header class="crayons-modal__box__header flag-user-modal-header">
+      <h2>Flag User</h2>
+      <button type="button" class="crayons-btn crayons-btn--icon crayons-btn--ghost modal-header-close-icon">
+        <svg width="24" height="24" viewBox="0 0 24 24" class="crayons-icon" xmlns="http://www.w3.org/2000/svg">
+          <path d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636l4.95 4.95z" />
+        </svg>
+      </button>
+    </header>
+    <div class="crayons-modal__box__body flag-user-modal-body">
+      <span>
+        Thanks for keeping DEV safe. Here is what you can do to flag this user:
```

Comment:
this should be generalized using `community_name` instead of `DEV`. it's a bit weird to have this much HTML in JavaScript. I would move it in a controller action and load it from JS